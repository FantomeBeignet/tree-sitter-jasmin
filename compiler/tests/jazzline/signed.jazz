#[signed=t]
fn _add (reg u64 v64) -> reg u64
requires #[prover=smt] { v64 <= 100 }
ensures #[prover=cas] { result.0 == 0}
{
reg u64 v64s;

#[tran = default] v64s = v64 + v64;

return v64s;
}