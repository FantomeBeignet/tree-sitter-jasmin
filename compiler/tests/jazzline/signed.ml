open Common

let () = parse_and_print "signed.jazz"  "_add"
