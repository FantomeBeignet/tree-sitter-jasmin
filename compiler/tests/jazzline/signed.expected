proc _add(v64s_135@sint64, v64_130@sint64) = 
{
 true && (v64_130@sint64 <= (const 64 (100)))
}
adds TMP_____146@uint1 v64s_135@sint64 v64_130@sint64 v64_130@sint64;
{
 /\[(v64s_135 = (0))] && true
}
 